### PR TITLE
Updated Validation Rules for Topics, Actions and Services

### PR DESCRIPTION
**Pull request by ipa-nhg-ha**

### CODE REVIEW

File: plugins/de.fraunhofer.ipa.ros.xtext/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/bin/
```
Comment (ipa-nhg) [inline plugins/de.fraunhofer.ipa.ros.xtext/.gitignore]: this file is not needed as there is a main one for the repository https://github.com/ipa320/ros-model/blob/master/.gitignore

File: plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/validation/RosValidator.xtend [hunk 2]
```diff
@@ -6,6 +6,11 @@ import org.eclipse.xtext.validation.Check
 import ros.Node
 import ros.Artifact
 import ros.Package
+import org.eclipse.emf.ecore.resource.Resource
```
Comment (ipa-nhg) [inline plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/validation/RosValidator.xtend]: I think this imports are not needed

File: plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/validation/RosValidator.xtend [hunk 3]
```diff
@@ -37,11 +42,83 @@ class RosValidator extends AbstractRosValidator {
...
 		}
- 	}}
+ 	}
+  }
+  
+  /* Customize Syntax Error Messages */
+  
+  /* For Topics */
+  @Check
+  def void CheckMsgsRefSubscriber(Node node){
```
Comment (ipa-nhg) [inline plugins/de.fraunhofer.ipa.ros.xtext/src/de/fraunhofer/ipa/ros/validation/RosValidator.xtend]: Sorry, this is not working as expected. It is marking as warning the full node. And so is quite hard to detect the line of the errors...

The checks have to be done (as my previous example) at interface level. That means:

```
@Check
def void CheckMsgsRefPublisher(Publisher pub){
     if (p.message.eContainer === null){
          warning("Couldn't resolve reference to TopicSpec \n" ... \n",
            null,INVALID_NAME)
      }
  	}
```

**Comment by ipa-nhg-ha** [comment]
Respected Miss Nadia

1) The imports for ros.node,ros.artifact, ros.package are necessary.
2) The file gitignore should i remove it from the project?
3) I will try the previous method. I have checked this one and you are right it is working on the full node and not where it should be.

Thanks and regards
Ameer Hamza


________________________________

**Comment by ipa-nhg** [comment]
Superseded by https://github.com/ipa320/ros-model/pull/83

**Pull request closed without merging**